sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -324,37 +350,75 @@ async def get_topic(self, topic_name: str, **kwargs) -> TopicDescription:
+        :keyword max_size_in_megabytes: The maximum size of the topic in megabytes, which is the size of
+         memory allocated for the topic.
+        :type max_size_in_megabytes: long
+        :keyword requires_duplicate_detection: A value indicating if this topic requires duplicate
+         detection.
+        :type requires_duplicate_detection: bool
+        :keyword duplicate_detection_history_time_window: ISO 8601 timeSpan structure that defines the
+         duration of the duplicate detection history. The default value is 10 minutes.
+        :type duplicate_detection_history_time_window: ~datetime.timedelta
+        :keyword enable_batched_operations: Value that indicates whether server-side batched operations
+         are enabled.
+        :type enable_batched_operations: bool
+        :keyword size_in_bytes: The size of the topic, in bytes.
+        :type size_in_bytes: int
+        :keyword filtering_messages_before_publishing: Filter messages before publishing.
+        :type filtering_messages_before_publishing: bool
+        :keyword is_anonymous_accessible: A value indicating if the resource can be accessed without
+         authorization.
+        :type is_anonymous_accessible: bool
+        :keyword authorization_rules: Authorization rules for resource.
+        :type authorization_rules:
+         list[~azure.servicebus.management.AuthorizationRule]
+        :keyword support_ordering: A value that indicates whether the topic supports ordering.
+        :type support_ordering: bool
+        :keyword auto_delete_on_idle: ISO 8601 timeSpan idle interval after which the topic is
+         automatically deleted. The minimum duration is 5 minutes.
+        :type auto_delete_on_idle: ~datetime.timedelta
+        :keyword enable_partitioning: A value that indicates whether the topic is to be partitioned
+         across multiple message brokers.
+        :type enable_partitioning: bool
+        :keyword enable_subscription_partitioning: A value that indicates whether the topic's
+         subscription is to be partitioned.
+        :type enable_subscription_partitioning: bool
+        :keyword enable_express: A value that indicates whether Express Entities are enabled. An express
+         queue holds a message in memory temporarily before writing it to persistent storage.
+        :type enable_express: bool
+        :keyword user_metadata: Metadata associated with the topic.
+        :type user_metadata: str
+
+        :rtype: ~azure.servicebus.management.TopicProperties
         """
-        try:
-            topic_name = topic.name  # type: ignore
-            to_create = topic._to_internal_entity()  # type: ignore
-        except AttributeError:
-            topic_name = topic  # type: ignore
-            to_create = InternalTopicDescription()  # Use an empty topic description.
+
+        topic = TopicProperties(name, **kwargs)
+        for key in topic.keys():
+            kwargs.pop(key, None)
```

Comment:
same comment as above, should this be moved into constructor?

sample concern applied to other model.

Reply:
Same reply as above